lib/taste_tester/host.rb

```diff
@@ -71,13 +72,22 @@ def runchef
       end
     end
 
+    def print_noop_warning
+      @PRINTED_WARNING ||= logger.warn(
```

Comment:
You can do that, but it's easier to do it the initializer for NoOp, and use a class variable (@@)